gx_suite_definitions/biomarkers.ipynb

```diff
@@ -0,0 +1,239 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import synapseclient\n",
+    "\n",
+    "import great_expectations as gx\n",
+    "import pandas as pd\n",
+    "import json\n",
+    "\n",
+    "context = gx.get_context(project_root_dir='../src/agoradatatools/great_expectations')\n",
+    "\n",
+    "from agoradatatools.gx import GreatExpectationsRunner\n",
+    "from expectations.expect_column_values_to_have_list_length import ExpectColumnValuesToHaveListLength\n",
+    "from expectations.expect_column_values_to_have_list_members import ExpectColumnValuesToHaveListMembers\n",
+    "from expectations.expect_column_values_to_have_list_members_of_type import ExpectColumnValuesToHaveListMembersOfType"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create Expectation Suite for Biomarkers Data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "syn = synapseclient.Synapse()\n",
+    "syn.login()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "biomarkers_data_file = syn.get(\"syn63540269\").path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Validator Object on Data File"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_json(biomarkers_data_file)\n",
+    "nested_columns = ['points']\n",
+    "df = GreatExpectationsRunner.convert_nested_columns_to_json(df, nested_columns)\n",
+    "validator = context.sources.pandas_default.read_dataframe(df)\n",
+    "validator.expectation_suite_name = \"biomarkers\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Add Expectations to Validator Object For Each Column"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# points\n",
+    "validator.expect_column_values_to_be_of_type(\"points\", \"str\")\n",
+    "with open(\"../src/agoradatatools/great_expectations/gx/json_schemas/immunohisto/points.json\", \"r\") as file:\n",
+    "    points_schema = json.load(file)\n",
+    "validator.expect_column_values_to_match_json_schema(\"points\", json_schema=points_schema)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# model\n",
+    "validator.expect_column_values_to_be_of_type(\"model\", \"str\")\n",
+    "validator.expect_column_values_to_not_be_null(\"model\")\n",
+    "# allows all alphanumeric characters, underscores, periods, and dashes\n",
+    "validator.expect_column_values_to_match_regex(\"type\", \"^[A-Za-z0-9\\s\\*_.-]+$\")\n"
```

Comment:
Just curious, are there any values that should NOT be allowed in this field?

Reply:
Lol, no. I just realized the regex got out of control in my attempt to capture all possible entries. Maybe I should just remove the regex validation haha

Reply:
Yeah I would say if there aren't any disallowed values, a regex isn't necessary.

Reply:
We could keep a regex here to ensure that we don't have the dreaded unknown unicode character; there is a discussion of this [on this PR](https://github.com/Sage-Bionetworks/agora-data-tools/pull/123/files) for team_info.description.

I don't think we've yet validated that this actually fails on the unknown unicode character, but it should.

Reply:
Actually, I take my above comment back,  I don't think we need the regex for this specific field.